cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/SparkProgramRunner.java

```diff
@@ -204,10 +210,17 @@ public ProgramController run(Program program, ProgramOptions options) {
       }
 
       boolean isLocal = SparkRuntimeContextConfig.isLocal(options);
-      SparkSubmitter submitter = isLocal
-        ? new LocalSparkSubmitter()
-        : new DistributedSparkSubmitter(hConf, locationFactory, host, runtimeContext,
-                                        options.getArguments().getOption(Constants.AppFabric.APP_SCHEDULER_QUEUE));
+      SparkSubmitter submitter;
+      // If MasterEnvironment is not available, use non-master env spark submitters
+      MasterEnvironment masterEnv = MasterEnvironments.getMasterEnvironment();
+      if (masterEnv != null) {
+        submitter = new MasterEnvironmentSparkSubmitter(locationFactory, host, runtimeContext, sparkConfig);
```

Comment:
need to get the sparkConfig from masterEnv and remove it from constructor.

Reply:
Just passing spark config since master env is just used for checking which submitter to use.